src/modules/previewpane/README.md

```diff
@@ -0,0 +1,175 @@
+
+
+<center>
+
+<figure>
+  <img src="../../../doc/images/preview_pane/markdown.gif" alt="Mark Down Preview Handler Demo" >
+  <figcaption>Figure 1 : Mark Down Preview Handler Demo</figcaption>
+</figure>
+
+</center>
+
+## Developing
+
+We have already done most of the development work in the [PreviewHandlerCommon](./common/cominterop/IPreviewHandler.cs) common project. To add a preview for the file type of .xyz:
+
+-  Add a new .NET project in the preview pane folder.
+-  Add a reference to the `PreviewHandlerCommon` common project.
+-  Create your preview handler class and extend the FileBasedPreviewHandler class. See an example below:
+
+```csharp
+using System;
+using System.Runtime.InteropServices;
+using Common;
+
+namespace XYZPreviewHandler
+{
+    /// <summary>
+    /// Implementation of preview handler for .xyz files.
+    /// GUID = CLSID / CLASS ID.
+    /// </summary>
+    [Guid("xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx")]
+    [ClassInterface(ClassInterfaceType.None)]
+    [ComVisible(true)]
+    public class XYZPreviewHandler : FileBasedPreviewHandler
+    {
+        private XYZPreviewHandlerControl xyzPreviewHandlerControl;
+
+        /// Call your rendering method here.
+        public override void DoPreview()
+        {
+            this.xyzPreviewHandlerControl.DoPreview(this.FilePath);
+        }
+
+        protected override IPreviewHandlerControl CreatePreviewHandlerControl()
+        {
+            this.xyzPreviewHandlerControl = new xyzPreviewHandlerControl();
+            return this.xyzPreviewHandlerControl;
+        }
+    }
+}
+```
+
+Create a separate Preview Handler Control class and extend the `FormHandlerControl` Class.
+
+```csharp
+using Common;
+
+namespace XYZPreviewHandler
+{
+    public class XYZPreviewHandlerControl : FormHandlerControl
+    {
+        public XYZPreviewHandlerControl()
+        {
+            // ... do your initializations here.
+        }
+
+        public override void DoPreview<T>(T dataSource)
+        {
+            // ... add your preview rendering code here.
+        }
+    }
+}
+```
+
+
+#### Integrate the Preview Handler into PowerToys Settings:
+
+Navigate to the [powerpreview](../previewpane/powerpreview/powerpreview.h) project and edit the `powerpreview.h` file. Add the following Settings Object instance to `m_previewHandlers` settings objects array in the constructor initialization:
+
+```cpp
+// XYZ Preview Handler Settings Object.
+FileExplorerPreviewSettings(
+    false,
+    L"<--YOUR_TOGGLE_CONTROL_ID-->",
+    L"<--A description of your preview handler-->",
+    L"xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx", // your preview handler CLSID.
+    L"<--A display name for your preview handler-->") 
+```
+
+## Install With MSIX
+
+<---To Do--->
```

Comment:
I might be mistaken but isn't there a PR for this? Are you still waiting on this?

Reply:
@udit3333  will create a separate PR for this section.